test/extended/openstack/utils.go

```diff
@@ -404,3 +405,17 @@ func isIpv4(ip string) bool {
 	}
 	return ipv4
 }
+
+// expand given ipv6 address
+func FullIPv6(ip net.IP) string {
+	dst := make([]byte, hex.EncodedLen(len(ip)))
+	_ = hex.Encode(dst, ip)
+	return string(dst[0:4]) + ":" +
+		string(dst[4:8]) + ":" +
+		string(dst[8:12]) + ":" +
+		string(dst[12:16]) + ":" +
+		string(dst[16:20]) + ":" +
+		string(dst[20:24]) + ":" +
+		string(dst[24:28]) + ":" +
+		string(dst[28:])
+}
```

Comment:
Oh wow. :smile: How about [this thing?](https://pkg.go.dev/net/netip#Addr.StringExpanded) This works in golang 1.20:

```
import (
	"fmt"
	"net/netip"
)

func main() {
	addr, _ := netip.ParseAddr("2001:db8::50")
	fmt.Println(addr.StringExpanded())
}
```

Reply:
I was looking fo something exactly like that :sweat:  Thank you!